src/main/kotlin/net/leanix/vsm/sbomBooster/service/BitBucketApiService.kt

```diff
@@ -0,0 +1,106 @@
+package net.leanix.vsm.sbomBooster.service
+
+import net.leanix.vsm.sbomBooster.configuration.PropertiesConfiguration
+import net.leanix.vsm.sbomBooster.domain.BitBucketAuthResponse
+import net.leanix.vsm.sbomBooster.domain.BitBucketRepositoriesResponse
+import net.leanix.vsm.sbomBooster.domain.GitProviderApiService
+import net.leanix.vsm.sbomBooster.domain.Repository
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Service
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.util.MultiValueMap
+import org.springframework.web.client.RestTemplate
+import java.util.*
+
+@Service
+class BitBucketApiService(
+    private val propertiesConfiguration: PropertiesConfiguration
+) : GitProviderApiService {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(BitBucketApiService::class.java)
+    }
+
+    override fun getUsername(token: String?): String? {
+        return propertiesConfiguration.bitbucketWorkspace
+    }
+
+    override fun getRepositories(token: String?, organization: String): List<Repository> {
+        val bearerToken = authenticate(token)
+
+        return getPaginatedRepositories(bearerToken, organization, null)
+    }
+
+    private fun getPaginatedRepositories(
+        bearerToken: String?,
+        organization: String,
+        pageUrl: String?
+    ): List<Repository> {
+        val restTemplate = RestTemplate()
+        val headers = HttpHeaders()
+
+        headers.set("Authorization", "Bearer $bearerToken")
+        val httpEntity: HttpEntity<MultiValueMap<String, String>> =
+            HttpEntity(headers)
+
+        val url = pageUrl ?: "https://api.bitbucket.org/2.0/repositories/$organization"
+
+        val responseEntity = restTemplate.exchange(
+            url, HttpMethod.GET, httpEntity,
+            BitBucketRepositoriesResponse::class.java
+        )
+        val bbRepositoriesResponse = responseEntity.body?.values ?: emptyList()
+
+        val repositories = mutableListOf<Repository>()
+        for (bbRepo in bbRepositoriesResponse) {
+            val cloneUrl =
+                bbRepo.links.clone.firstOrNull { it.name == "https" }?.href?.replaceFirst("[^/]+@".toRegex(), "")
+            val sourceInstance = if (propertiesConfiguration.sourceInstance.isBlank())
+                propertiesConfiguration.bitbucketWorkspace
+            else
+                propertiesConfiguration.sourceInstance
+
+            repositories.add(
+                Repository(
+                    cloneUrl.orEmpty(),
+                    propertiesConfiguration.sourceType,
+                    sourceInstance,
+                    bbRepo.name
```

Comment:
Hi @dsimpson-leanix,
We decided to also include the repoId for debugging purposes.
Could you add this small change here and any other place that is misssing?
Please refer to: https://github.com/leanix/vsm-sbom-booster/pull/36

Reply:
This may be easier to merge if I wait for #36 to be released, then rebase my branch on that commit.

Reply:
Added